otel-worker-cli/src/commands/mcp.rs

```diff
@@ -285,78 +400,91 @@ pub enum Transport {
+async fn handle_client_message(
+    state: &McpState,
+    session: &McpSession,
+    client_message: ClientMessage,
+) {
     match client_message {
         ClientMessage::Request(request) => {
-            handle_client_request(state, request).await;
+            handle_client_request(state, session, request).await;
         }
         ClientMessage::Notification(notification) => {
-            handle_client_notification(notification).await;
+            handle_client_notification(state, session, notification).await;
         }
         ClientMessage::Response(response) => {
-            handle_client_response(response).await;
+            handle_client_response(state, session, response).await;
         }
         ClientMessage::Error(error) => {
-            handle_client_error(error).await;
+            handle_client_error(state, session, error).await;
         }
     }
 }
 
-async fn handle_client_request(state: &McpState, request: ClientJsonrpcRequest) {
-    let result: Result<ResultFromServer> = match request.request {
+async fn handle_client_request(
+    state: &McpState,
+    session: &McpSession,
+    request: ClientJsonrpcRequest,
+) {
+    let request_id = request.id.clone();
+
+    let result = match request.request {
         RequestFromClient::ClientRequest(client_request) => match client_request {
             ClientRequest::InitializeRequest(inner_request) => {
-                handle_initialize(state, inner_request.params)
-                    .await
-                    .map(Into::into)
+                handle_initialize(state, &session, request.id, inner_request.params).await
             }
             ClientRequest::ListResourcesRequest(inner_request) => {
-                handle_resources_list(state, inner_request.params)
-                    .await
-                    .map(Into::into)
+                handle_resources_list(state, &session, request.id, inner_request.params).await
             }
             ClientRequest::ReadResourceRequest(inner_request) => {
-                handle_resources_read(state, inner_request.params)
-                    .await
-                    .map(Into::into)
+                handle_resources_read(state, &session, request.id, inner_request.params).await
+            }
+            ClientRequest::PingRequest(inner_request) => {
+                handle_ping(state, &session, request.id, inner_request.params).await
             }
-            ClientRequest::PingRequest(inner_request) => handle_ping(state, inner_request.params)
-                .await
-                .map(Into::into),
             _inner_request => {
-                error!(method = request.method, "Received unsupported requests");
-                return;
+                error!(
+                    method = request.method,
+                    "received unsupported json-rpc message"
+                );
+                Err(anyhow!("received unsupported json-rpc message"))
```

Comment:
`Err(anyhow!(""))` can be shortened to just `bail!("")`

Reply:
In this case that doesn't work, since I want to return a `Err(...)` for the next part of the code. `bail!` adds a `return ...` which will stop the entire function.